Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -722,7 +825,33 @@ extension NIOSSLHandler {
 extension NIOSSLHandler {
     private typealias BufferedWrite = (data: ByteBuffer, promise: EventLoopPromise<Void>?)
 
+    /// Completes the given promise when all outstanding writes have been flushed.
+    private func bufferedWritesCascadeFlushPromise(
+        to promise: EventLoopPromise<Void>?,
```

Comment:
I don't think this use of the promise is correct. It should be attached to the `close(mode: .output)` operation. We should _fail_ it if we don't actually issue that call, but we shouldn't succeed it ourselves.